examples/websockets/src/client.rs

```diff
@@ -0,0 +1,165 @@
+//! Based on tokio-tungstenite example websocket client, but with multiple
```

Comment:
Hm I'm kinda torn on having a client example here. Like what is it trying to demonstrate? How to write a websocket client in rust? Maybe just linking to [tungstenite's examples](https://github.com/snapview/tungstenite-rs/tree/master/examples) is enough for that. I generally prefer axum's examples to focus on axum related things, which I'd argue this isn't.

Reply:
TL;DR version:
- tungstenite example is very "shallow" and only shows how *one* client would interact with the server, which is already demonstrated well enough by the browser example.
- I wanted specifically do demonstrate how one could lay groundwork for more complex/non-standard state machines with multiple concurrent flows, rather than the beaten to death chat example. It would be tough to demonstrate with a silly client.

The entire idea I had with conceptual improvements to the example  was to really show good reasons one would want to use Axum implementation in particular if compared to, for example, actix. For me personally the key reason is specifically how one can have independent send and receive logic for the websockets, and how all of that can be 100% async-aware, and with minimal boilerplate. For contrast, in Actix the expectation is that one either

-  handles messages as "copy types", which means they are fully buffered before feeding them into a handler using their Actor model, or, alternatively, 
- or uses their low-level API, which is "fun", and you'd have to handle things like ping/pong yourself which is ugly. [https://github.com/actix/examples/blob/master/websockets/chat-actorless/src/handler.rs#L54](as they show in their own examples)

Reply:
Alright I guess that fair. I'm okay with keeping it but I still my earlier comments about trimming it down (removing argument parsing etc) make sense. wdyt?

Reply:
totally agree, will address them once I have some spare time.